packages/next/server/lib/error-overlay-middleware.js

```diff
@@ -1,12 +1,19 @@
 import url from 'url'
 import launchEditor from 'launch-editor'
+import path from 'path'
+
+export default function errorOverlayMiddleware(req, res, next) {
+
+  const srcRoot = process.cwd()
 
-export default function errorOverlayMiddleware (req, res, next) {
   if (req.url.startsWith('/_next/development/open-stack-frame-in-editor')) {
     const query = url.parse(req.url, true).query
     const lineNumber = parseInt(query.lineNumber, 10) || 1
     const colNumber = parseInt(query.colNumber, 10) || 1
-    launchEditor(`${query.fileName}:${lineNumber}:${colNumber}`)
+
+    const resolvedFileName = path.join(srcRoot, query.fileName)
```

Comment:
Sometimes, `query.fileName` _may_ be a full path. To cover this case, we should probably use `resolve` over `join` so the `srcRoot` is disregarded in this case.
```suggestion
    const resolvedFileName = path.resolve(srcRoot, query.fileName)
```

Reply:
Imo we should always use `join` because `resolve` handles weird behaviors 😅

Reply:
After a bit of digging around, I've seen(maybe I'm wrong), that the `resolvedFilename` from https://github.com/zeit/next.js/blob/163830c026f61370b9c1f5d05005f52c91e7ecee/packages/next/client/dev-error-overlay/hot-dev-client.js#L55

is always "absolute", like:

```
/a
 --/b
 ----/c
 ------Carousel.js
/pages
```
![image](https://user-images.githubusercontent.com/4474353/52814257-adf93300-3069-11e9-874f-9e8364319023.png)

```
/pages
Carousel.js
```

![image](https://user-images.githubusercontent.com/4474353/52814364-f3b5fb80-3069-11e9-9d4c-3894054b842e.png)


So, for example
```js
const resolvedFileName = path.resolve(process.cwd(), '/Carousel.js')
// yields: `/Carousel.js`

const resolvedFileName = path.resolve(process.cwd(), '/a/b/c/Carousel.js')
// yields: `/a/b/c/Carousel.js`
```

Because it already thinks that the second argument is absolute and disregards the first (from [path.resolve() docs](https://nodejs.org/api/path.html#path_path_resolve_paths))

So I think I'll add this:
```js
 let resolvedFileName = query.fileName;

 if (!fs.existsSync(resolvedFileName)) {
    resolvedFileName = path.join(process.cwd(), resolvedFileName);
 }
```

Reply:
I think it's a bad idea to use `process.cwd()` btw, we should use the `dir` that's from Next.js.

Reply:
Hi @timneutkens :)

Is this the right place to like pass down the `dir` value?
https://github.com/zeit/next.js/blob/163830c026f61370b9c1f5d05005f52c91e7ecee/packages/next/server/hot-reloader.js#L151

Reply:
Hey @klujanrosas I missed your comment, the correct way to do it is to pass it in here:

https://github.com/zeit/next.js/blob/163830c026f61370b9c1f5d05005f52c91e7ecee/packages/next/server/hot-reloader.js#L256

Basically change it to 

```
errorOverlayMiddleware({
  dir: this.dir
}),
```

And then change the middleware:

https://github.com/zeit/next.js/blob/163830c026f61370b9c1f5d05005f52c91e7ecee/packages/next/server/lib/error-overlay-middleware.js#L4

To be like this:

```
export default function errorOverlayMiddleware (options) {
  return (req, res, next) => {
	  if (req.url.startsWith('/_next/development/open-stack-frame-in-editor')) {
	    const query = url.parse(req.url, true).query
	    const lineNumber = parseInt(query.lineNumber, 10) || 1
	    const colNumber = parseInt(query.colNumber, 10) || 1
	    launchEditor(`${query.fileName}:${lineNumber}:${colNumber}`)
	    res.end()
	  } else {
	    next()
	  }
  }
}
```

Reply:
Then you have `options.dir` available 👍